### PR TITLE
#2877 kinsey bellerose get singular graph endpoint

### LINKED ISSUE #2877: [Stats Page] - Create Get Single Graph Endpoint
### Description

We need to create an endpoint to get a graph. This endpoint should be in the statistics router under the route ‘statistics/:graphId’ and should take in a graph id.

Be sure to include unit tests and screenshots of postman manual tests to ensure everything is working correctly.

### Acceptance Criteria

- a new endpoint is created in the statistics router under the route ‘statistics/:graphId’ that returns a single graph.
- unit tests and postman screenshot

### Proposed Solution

Take a look at some of the other get endpoints we have on Finishline and use it as a reference for this endpoint.

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by cielbellerose**
## Changes

Created the endpoint for getting a single graph

## Test Cases

- Valid graph id
- Invalid graph id 

## Screenshots

Screenshot 2024-12-10 at 11 15 22 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #2877

### CODE REVIEW

File: src/backend/src/services/statistics.services.ts [hunk 1]
```diff
@@ -115,4 +117,19 @@ export default class StatisticsService {
       };
     });
   }
+
+  static async getSingleGraph(id: string, organization: Organization): Promise<Graph> {
+    // if (!(await userHasPermissionNew(user.userId, organization.organizationId, ['VIEW_GRAPH']))) {
```
Comment (Peyton-McKee) [inline src/backend/src/services/statistics.services.ts]: i think you need to uncomment this.

File: src/backend/src/services/statistics.services.ts [hunk 2]
```diff
@@ -115,4 +117,15 @@ export default class StatisticsService {
       };
     });
   }
+
```
Comment (Peyton-McKee) [inline src/backend/src/services/statistics.services.ts:124]: can you add javadoc here. basically if you do /** before a function it should autogenerate with the parameters you need to fill in.

File: src/backend/src/services/statistics.services.ts [hunk 3]
```diff
@@ -115,4 +117,15 @@ export default class StatisticsService {
...
+  static async getSingleGraph(id: string, organization: Organization): Promise<Graph> {
+    const requestedGraph = await prisma.graph.findUnique({
```
Comment (Peyton-McKee) [inline src/backend/src/services/statistics.services.ts:135]: you removed the comment, but I think we do want that permission check actually.

File: src/backend/src/services/statistics.services.ts [hunk 4]
```diff
@@ -115,4 +117,15 @@ export default class StatisticsService {
...
+      where: { id, organizationId: organization.organizationId },
+      ...getGraphQueryArgs(organization.organizationId)
+    });
+
+    if (!requestedGraph) throw new NotFoundException('Graph', id);
+
```
Comment (Peyton-McKee) [inline src/backend/src/services/statistics.services.ts:152]: can you add a check if the graph is deleted as well.

File: src/backend/src/services/statistics.services.ts [hunk 5]
```diff
@@ -115,4 +117,15 @@ export default class StatisticsService {
...
+    return graphTransformer(requestedGraph as Prisma.GraphGetPayload<GraphQueryArgs> & { graphData: GraphData[] });
```
Comment (Peyton-McKee) [inline src/backend/src/services/statistics.services.ts]: you shouldn’t have to use an as here. if your missing the graphData you probably need to retrieve that. which you should be able to use by using the getGraphData function above

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
Looking good so far!

File: src/backend/src/services/statistics.services.ts (hunk 5 above)
Comment (cielbellerose) [reply src/backend/src/services/statistics.services.ts]: Screenshot 2024-12-12 at 12 22 18 PM
fixed the 'as', but I don't fundamentally understand the query path - for GraphGen its a single QueryPath but the Graph schema has queryPaths Graph_Query[]

Comment (Peyton-McKee) [reply src/backend/src/services/statistics.services.ts]: theres a comment in the schema explaining how that works, but i think were gonna get rid of that actually

Comment (cielbellerose) [reply src/backend/src/services/statistics.services.ts]: tried my best to convert, however I'm having this new issue where none of the stat tests (even the ones I didn't write) aren't passing Screenshot 2024-12-12 at 7 48 02 PM

Comment (cielbellerose) [reply src/backend/src/services/statistics.services.ts]: tried my best to convert - not sure how to go about fixing the test that failed

**Review by Peyton-McKee — APPROVED** [review]

**Pull request merged**